### PR TITLE
tests/periph_rtt_min: adapt for slower RTTs

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
The test still occasionally fails in the nightlies which points to a too short timeout.

### Testing procedure
We maybe should give Murdock a few spins with that to see if the error reoccurs there.

### Issues/PRs references
None

### CODE REVIEW

**Comment by fjmolinas** [comment]
I think increasing the timeout is not the best fix since it might still fail with slow RTTs, how about adapting to something like (untested):

```python
    test_end = r'RTT_MIN_OFFSET for [a-zA-Z\-\_0-9]+ over {samples} samples: \d+' .format(samples=exp_samples)
    test_ongoing = r'.'
    while child.expect([test_ongoing, test_end]) == 0:
        pass
```

We won't count the dots but just use them as a parameter that the test is running. What do you think?

**Comment by miri64** [comment]
> I think increasing the timeout is not the best fix since it might still fail with slow RTTs, how about adapting to something like (untested):
>
> ```python
>     test_end = r'RTT_MIN_OFFSET for [a-zA-Z\-\_0-9]+ over {samples} samples: \d+' .format(samples=exp_samples)
>     test_ongoing = r'.'
>     while child.expect([test_ongoing, test_end]) == 0:
>         pass
> ```
> 
> We won't count the dots but just use them as a parameter that the test is running. What do you think?

I think in that case we should not print flushed dots but just some line. This spams the output somewhat, but at least we do not have to rely on `fflush` really working.

On the other hand, do you really think a slow RTT is the problem here? The RTT should be configured as on all other `samr21-xpro`s and locally it works. For a slow RTT we already have the fix of decreasing the `SAMPLES` for the test.

**Comment by fjmolinas** [comment]
> > I think increasing the timeout is not the best fix since it might still fail with slow RTTs, how about adapting to something like (untested):
> > ```python
> >     test_end = r'RTT_MIN_OFFSET for [a-zA-Z\-\_0-9]+ over {samples} samples: \d+' .format(samples=exp_samples)
> >     test_ongoing = r'.'
> >     while child.expect([test_ongoing, test_end]) == 0:
> >         pass
> > ```
> > 
> > 
> > We won't count the dots but just use them as a parameter that the test is running. What do you think?
> 
> I think in that case we should not print flushed dots but just some line. This spams the output somewhat, but at least we do not have to rely on `fflush` really working.

Since we don't match the amount of dots, it doesn't matter if some are missed. But its a test so if you think its better to spam the output go ahead.

> On the other hand, do you really think a slow RTT is the problem here? The RTT should be configured as on all other `samr21-xpro`s and locally it works. For a slow RTT we already have the fix of decreasing the `SAMPLES` for the test.

I don't think its the case for this `BOARD` (`samr21-xpro`), no, but some `BOARD`s have the RTT running at 1Hz, this is actually the reason why I counted dots in the initial `periph/rtt_min` test (yes too bad I just now remembered, and yes I should have added a comment on that to not miss it when looking at #16314).

**Comment by miri64** [comment]
> > On the other hand, do you really think a slow RTT is the problem here? The RTT should be configured as on all other `samr21-xpro`s and locally it works. For a slow RTT we already have the fix of decreasing the `SAMPLES` for the test.
> 
> I don't think its the case for this `BOARD` (`samr21-xpro`), no, but some `BOARD`s have the RTT running at 1Hz, this is actually the reason why I counted dots in the initial `periph/rtt_min` test (yes too bad I just now remembered, and yes I should have added a comment on that to not miss it when looking at #16314).

Still: why not just decrease the number of dots then?

**Comment by fjmolinas** [comment]
> > > On the other hand, do you really think a slow RTT is the problem here? The RTT should be configured as on all other `samr21-xpro`s and locally it works. For a slow RTT we already have the fix of decreasing the `SAMPLES` for the test.
> > 
> > 
> > I don't think its the case for this `BOARD` (`samr21-xpro`), no, but some `BOARD`s have the RTT running at 1Hz, this is actually the reason why I counted dots in the initial `periph/rtt_min` test (yes too bad I just now remembered, and yes I should have added a comment on that to not miss it when looking at #16314).
> 
> Still: why not just decrease the number of ~dots~ samples then?

Many samples are required to get the actual value, why 1000? I don't remember, probably what worked well from trying on many BOARDs.

If you mean decreasing the number of samples per BOARD, I find it better that the test script Is robust against the application timings or delays instead of adjusting the application to fit the test script.

**Comment by miri64** [comment]
Fine. Given that the test timed out as before, I applied the fix as suggested.

File: tests/periph_rtt_min/tests/01-run.py [hunk 1]
```diff
@@ -15,10 +15,11 @@ def testfunc(child):
...
-        .format(samples=exp_samples)
-    )
+    test_end = r'RTT_MIN_OFFSET for [a-zA-Z\-\_0-9]+ over {samples} ' \
+               r'samples: \d+'.format(samples=exp_samples)
+    test_ongoing = r'Sample \d+'
+    while child.expect([test_end, test_ongoing]):
+        pass
 
 
 if __name__ == "__main__":
```
Comment (miri64) [inline tests/periph_rtt_min/tests/01-run.py:25]: If we want to reduce the spam, we can also set `echo=False` in the line below.

**Comment by fjmolinas** [comment]
I ran the test and it passed on all BOARDs buy [atmega256rfr2-xpro](https://ci.inria.fr/ci-riot-tribe/job/build-pipeline.jk/264/artifact/atmega256rfr2-xpro/tests/periph_rtt_min/test.failed/*view*/), I'll take a look at whats going on there.

**Comment by fjmolinas** [comment]
> I ran the test and it passed on all BOARDs buy [atmega256rfr2-xpro](https://ci.inria.fr/ci-riot-tribe/job/build-pipeline.jk/264/artifact/atmega256rfr2-xpro/tests/periph_rtt_min/test.failed/*view*/), I'll take a look at whats going on there.

Same issue in master, so lets go ahead  with this one, please squash @miri64

**Comment by miri64** [comment]
Mhhh seems like this did not fix the problem. Can it happen that the RTT hangs?

**Comment by fjmolinas** [comment]
How about adding a REMOVEME to `RIOT/.murdock` to only build test this application on samr21-xpro?

**Comment by miri64** [comment]
> How about adding a REMOVEME to `RIOT/.murdock` to only build test this application on samr21-xpro?

I did this yesterday ;-). With that the test passed every time (I re-did the test about 5-10 times).

**Comment by miri64** [comment]
> I did this yesterday ;-). With that the test passed every time (I re-did the test about 5-10 times).

For some reason Github does not show all the `CI: ready for build` label changes ;-).

**Comment by miri64** [comment]
But see https://github.com/RIOT-OS/RIOT/compare/27ff2252e89dd24188f70ea208c312f04c9b28c5..db1533d55db95f4f494e3631fb1dd787213fef4a.

**Comment by fjmolinas** [comment]
> But see https://github.com/RIOT-OS/RIOT/compare/27ff2252e89dd24188f70ea208c312f04c9b28c5..db1533d55db95f4f494e3631fb1dd787213fef4a.

Ah nice, sorry I missed that.

**Comment by miri64** [comment]
> Mhhh seems like this did not fix the problem. Can it happen that the RTT hangs?

Confirmed locally that the `samr21-xpro` at `pi-c06df873` is just b0rken. I connected a new `samr21-xpro` so the problem should now be fixed.

> > I ran the test and it passed on all BOARDs buy [atmega256rfr2-xpro](https://ci.inria.fr/ci-riot-tribe/job/build-pipeline.jk/264/artifact/atmega256rfr2-xpro/tests/periph_rtt_min/test.failed/*view*/), I'll take a look at whats going on there.
> 
> Same issue in master, so lets go ahead with this one, please squash @miri64

Done, in case we still want to go ahead with this :-).

**Review by fjmolinas — APPROVED** [review]
ACK, builds are passing again, the test samr21 was at fault

**Pull request merged**